arches/app/src/components/ControlledListManager/ControlledListInventory.vue

```diff
@@ -0,0 +1,196 @@
+
+const items: Ref<ControlledList[]> = ref([]);
+const toast = useToast();
+const { $gettext, $ngettext } = useGettext();
+const lightGray = "#f4f4f4";
+
+const fetchLists = async () => {
+    let errorText;
+    try {
+        const response = await fetch(arches.urls.controlled_lists);
+        if (!response.ok) {
+            errorText = response.statusText;
+            const body = await response.json();
+            errorText = body.message;
+            throw new Error();
+        } else {
+            await response.json().then((data) => {
+                props.setLanguageMap(data.languages);
+                items.value = data.controlled_lists;
+            });
+        }
+    } catch {
+        toast.add({
+            severity: "error",
+            summary: errorText || $gettext("Unable to fetch lists"),
+            life: 3000,
+        });
+    }
+};
+
+const createList = async () => {
+    try {
+        const response = await fetch(arches.urls.controlled_list_add, {
+            method: "POST",
+            headers: {
+                "X-CSRFToken": Cookies.get("csrftoken"),
+            },
+        });
+        if (response.ok) {
+            const newItem = await response.json();
+            items.value.unshift(newItem);
+        } else {
+            throw new Error();
+        }
+    } catch {
+        toast.add({
+            severity: "error",
+            summary: $gettext("List creation failed"),
+            life: 3000,
+        });
+    }
+};
+
+const deleteLists = async (selectedItems: ControlledList[]) => {
+    if (!selectedItems.length) {
+        return;
+    }
+    const promises = selectedItems.map((list) =>
+        fetch(arches.urls.controlled_list(list.id), {
+            method: "DELETE",
+            headers: {
+                "X-CSRFToken": Cookies.get("csrftoken"),
+            },
+        })
+    );
+
+    const shouldResetDisplay = (
+        props.displayedList && selectedItems.includes(props.displayedList)
+    );
+
+    try {
+        const responses = await Promise.all(promises);
+        if (responses.some((resp) => resp.ok)) {
+            if (shouldResetDisplay) {
+                props.setDisplayedList(null);
+            }
+        }
+        if (responses.some((resp) => !resp.ok)) {
+            throw new Error();
+        }
+    } catch {
+        toast.add({
+            severity: "error",
+            summary: $gettext("One or more lists failed to delete."),
+            life: 3000,
+        });
+    }
+    await fetchLists();
+};
+</script>
+
+<template>
+    <Splitter
+        :pt="{
+            gutter: { style: { background: lightGray } },
+            gutterHandler: { style: { background: lightGray } },
+        }"
+    >
+        <SplitterPanel
+            :size="30"
+            :min-size="15"
+        >
+            <div class="header">
+                <h2>{{ $gettext('Controlled Lists') }}</h2>
+            </div>
+
+            <Suspense>
+                <SidepanelDataView
```

Comment:
I get the idea of geneaicizing a component like `SidePanelDataView`. However I think it may end up being better if instead this is replaced with single-purpose components.

However this isn't a hard && fast opinion and I'm happy to chat about the different patterns.

Reply:
I agree, reusable high-level components have been pretty rare in my experience. However, when I saw how similar the UI was between these two, and when I heard that we may reuse some of these components in the ADM, I was tempted to try. One obvious tradeoff is the [other thread where this came up](https://github.com/archesproject/arches/pull/10569#discussion_r1496188599)